src/App.tsx

```diff
@@ -1,223 +1,225 @@
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterOption, setFilterOption] = useState(Filters.ALL);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [isAdded, setIsAdded] = useState(false);
+  const [deletedTodoID, setDeletedTodoID] = useState<number[]>([]);
+  const [temporaryTodo, setTemporaryTodo] = useState<Todo | null>(null);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    if (user) {
+      getTodos(user.id)
+        .then(setTodos)
+        .catch(() => setErrorMessage('Unable to load a todos'));
+    }
+  }, [user?.id]);
+
+  const addNewTodo = useCallback((event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!newTodoTitle) {
+      setErrorMessage('Title can\'t be empty');
+
+      return;
+    }
+
+    const newTodoAdding = async () => {
+      setIsAdded(true);
+
+      if (user) {
+        setTemporaryTodo({
+          id: 0,
+          userId: user?.id,
+          title: newTodoTitle,
+          completed: false,
+        });
+
+        try {
+          const newTodo = await addTodo({
+            userId: user.id,
+            title: newTodoTitle,
+            completed: false,
+          });
+
+          setTemporaryTodo(null);
+
+          setTodos(currentTodos => [...currentTodos, newTodo]);
+
+          setNewTodoTitle('');
+        } catch (e) {
+          setErrorMessage('Unable to add a todo');
+        } finally {
+          setIsAdded(false);
+        }
+      }
+    };
+
+    newTodoAdding();
+  }, [newTodoTitle]);
+
+  const deleteTodo = useCallback(async (todoId: number) => {
+    setDeletedTodoID(prev => [...prev, todoId]);
+    try {
+      await removeTodo(todoId);
+
+      setTodos(currentTodos => currentTodos.filter(
+        todo => todo.id !== todoId,
+      ));
+    } catch (e) {
+      setErrorMessage('Unable to delete a todo');
+    } finally {
+      setDeletedTodoID(prev => {
+        return prev.filter(deletedID => deletedID !== todoId);
+      });
+    }
+  }, []);
+
+  const deleteAllComplitedTodos = useCallback(async () => {
+    const completedTodoId = todos
+      .filter(todo => todo.completed)
+      .map(todo => todo.id);
+
+    setDeletedTodoID(completedTodoId);
+
+    await Promise.all(completedTodoId.map(id => deleteTodo(id)));
+    setTodos(prev => prev.filter(todo => !completedTodoId.includes(todo.id)));
+
+    setDeletedTodoID([0]);
+  }, [todos]);
+
+  const changeTodo = useCallback(async (
+    todoId: number,
+    fieldsToUpdate: Partial<Todo>,
+  ) => {
+    setDeletedTodoID(prev => [...prev, todoId]);
+    try {
+      await updateTodo(
+        todoId,
+        fieldsToUpdate,
+      );
+
+      setTodos(currentTodos => currentTodos.map(todo => (
+        todo.id === todoId
+          ? { ...todo, ...fieldsToUpdate }
+          : todo
+      )));
+    } catch (error) {
```

Comment:
Do you use this error?
```suggestion
    } catch  {
```

Reply:
done